app/components/common/BaseVideo.vue

```diff
@@ -0,0 +1,54 @@
+<script>
+
+import BaseCloudflareVideo from './BaseCloudflareVideo'
+
+const TYPES = {
+  YOUTUBE: 'youtube',
+  CLOUDFLARE: 'cloudflare'
+}
+
+export default {
+  components: {
+    BaseCloudflareVideo
+  },
+
+  props: {
+    youtubeProps: {
+      type: Object
+    },
+    cloudflareProps: {
+      type: Object
+    },
+    defaultType: {
+      type: String,
+      default: TYPES.YOUTUBE
+    }
+  },
+
+  data () {
+    return {
+      activeType: this.defaultType,
+      TYPES
+    }
+  },
+  methods: {
+    youtubeError () {
+      this.activeType = TYPES.CLOUDFLARE
+    }
+  }
+}
+</script>
+
+<template>
+  <div class="video-div">
+    <youtube
+        v-if="activeType==TYPES.YOUTUBE"
+        @error="youtubeError"
+        v-bind="youtubeProps"
+    />
```

Comment:
This could be future work, but can we somehow get a `?rel=0` parameter here, or other way to specify this, so that it shows related videos from our channel instead of all YouTube after it is done playing?

Bonus future work would be to adapt some public script for listening to playback end and rewinding to beginning and/or putting a playback over screen above the video, to not show any related videos at all.